src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
-export interface Figure {
+type FigureShape = 'triangle' | 'circle' | 'rectangle';
+type FigureColor = 'red' | 'green' | 'blue';
+type AreaCallback = () => number;
 
+export interface Figure {
+  shape: FigureShape,
+  color: FigureColor,
+  getArea: AreaCallback,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: FigureShape = 'triangle';
+
+  constructor(
+    public color: FigureColor,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a * b * c <= 0) {
+      throw new Error('You have entered incorrect data');
+    }
+
+    if (a + b <= c || b + c <= a || a + c <= b) {
+      throw new Error(`sides ${a} ${b} ${c} can't form a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const sp = (this.a + this.b + this.c) / 2;
+    const aDifference = sp - this.a;
+    const bDiffernece = sp - this.b;
+    const cDifference = sp - this.c;
+    const area = Math.sqrt(sp * aDifference * bDiffernece * cDifference);
 
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Circle {
+export class Circle implements Figure {
+  shape: FigureShape = 'circle';
 
+  constructor(
+    public color: FigureColor,
+    public radius: number,
+  ) {
+    if (radius < 0) {
+      throw new Error('You have entered incorrect data');
```

Comment:
Give more descriptive error message
P.S. Also, would be a good idea to create variables for all error messages, some of them can be reused